### ISSUE TITLE
Unable to re-authenticate device after reinstalling tailscale 

### ISSUE THREAD

**Issue by domenicquirl** [tailscale/tailscale#9382]
### What is the issue?

We (as a company) recently switched to `tailscale` and I am currently in the process of migrating our provisioning infrastructure to install `tailscale` on our devices. To test the changes, I ran the provisioning process several times with a (the same) testing device. At some point, I appeared to suddenly be unable to repeat the `tailscale up` part of the setup process even though I had not changed anything related to the way this part runs, failing with the error message `backend error: invalid key: API key XXXXXXX not valid` for a key I had just generated in the admin console.

I was since able to figure out that the reason this happens is that there is something called a "machine identity" which lives in `/var/lib/tailscale/tailscaled.state` and does not change with un- / reinstalls. However, finding that out required finding [this Reddit thread](https://www.reddit.com/r/Tailscale/comments/yrs9tt/unable_to_authenticate_after_cloning_linux_vm/) where a comment was made on a similar issue occurring with cloned VMs. The process of getting there was extremely confusing for several reasons:

1. As pointed out by the OP u/Pielander29 on Reddit, the error message calls the authentication key an API key, while API _access tokens_ are a different thing but generated right next to auth keys in the settings.
2. The admin console's configuration logs show the device being added. They show the auth key being created and revoked, but it is expected for a one-time key to be revoked upon use and they also clearly state that the node was created, its settings set, that I logged in the node and that it was approved. Why would a node be approved if the key is invalid? And why does a node that was just created, logged in and approved not show up as a machine in the console? There is no mention in the logs of the node being removed again.
3. The error message says "invalid key" even though the key is definitely valid and I just generated it as the owner of the tailnet.
4. The login command is unchanged from before and the device is not currently part of the network, so it's not obvious why I would be unable to add it.

I am opening this as a bug report because I expected to be able to connect the device. I can also understand if you have a different view on that (in particular as I have found no other mention of the "machine identity", what it represents and what it is used for in `tailscale`), but at least the error message and logs could give a much better indication of what the underlying issue is (I don't know what `tailscale` considers bad diagnostics as in terms of issues).

### Steps to reproduce

1. Install `tailscale` on a Linux (in our case Raspbian) device via the official installer / script.
2. Authenticate device via `tailscale up --hostname=... --authkey=... --ssh` using a pre-approved authkey
3. Remove device from tailnet via web console
4. Uninstall `tailscale` (and possibly `tailscale-archive-keyring`) via `apt-get remove`. At this point neither `tailscale` nor `tailscaled` exist on the system.
5. Repeat step (1).
6. Repeat step (2) - fails with `backend error: invalid key: API key XXXXXXX not valid`

### OS

Linux

### OS version

Raspbian stretch

### Tailscale version

1.48.2

**Comment by DentonGentry** [comment]
The presence or absence of a `tailscaled.state` file would not result in `backend error: invalid key: API key XXXXXXX not valid`.

The intent of filing this issue is to diagnose why the authkey failed to authenticate the device? Most commonly, this would be a single-use authkey which was already been used once.

**Comment by domenicquirl** [comment]
No, as stated in the issue description and described in the Reddit comment (I assume the comment is from someone from your team?) the error was resolved by removing the file and having it be regenerated with the new installation. I am thus quite confident that the error is indeed related to the presence of that file. The error message does not indicate that in any way, so I understand the confusion, but that's kinda exactly the problem. 

If, as you say, the presence or absence of the identity file _shouldn't_ result in this error, then that's a different bug because that's precisely what seems to be happening 😅 

Either way, as I also said in the description the key was fresh and definitely valid. In fact, I tested this with multiple newly generated keys associated with both my account and that of a colleague who is also a tailscale Admin, though admittedly only with single-use keys, so I don't know what happens if you use a reusable key instead. I assume it would be similar, as to me it seems the key is not the actual issue here.

**Comment by bradkehler** [comment]
To update the sourced reddit comment, in DSM 7.2.2 the state file that needs to be deleted is now in /volume1/@appdata/Tailscale/tailscaled.state assuming that the app was/is installed on volume1.

**Comment by 369Martin369** [comment]
I have same problem. my NAS was once bound to a another user. deleted it there and changed ID. Then I cant join new user, as nodekey already exists . Obviously the nodekey was not removed when I removed the device in the admin panel. thats a bug

**Comment by sovereignintegritysystems** [comment]
If a fresh login is rejected and the daemon keeps regenerating a node key every few seconds, the usual cause is the stale machine identity persisted in `/var/lib/tailscale/tailscaled.state` from the previous registration. `tailscale login` (not `tailscale up`) cleared it for me on NixOS — it starts a clean profile and sidesteps the stale record (the documented #9025 workaround). If `login` alone still loops, resetting the state file is the logical next step:

```
sudo systemctl stop tailscaled
sudo rm -f /var/lib/tailscale/tailscaled.state
sudo systemctl start tailscaled
sudo tailscale login
```
